paddle/fluid/operators/listen_and_serv_op.cc

```diff
@@ -29,6 +29,28 @@ void RunServer(std::shared_ptr<detail::AsyncGRPCServer> service) {
   VLOG(4) << "RunServer thread end";
 }
 
+namespace {
+
+bool global_exit_flag = false;
+
+typedef std::unordered_set<std::shared_ptr<detail::ReceivedQueue>>
+    BlockingQueueSet;
+BlockingQueueSet global_blocking_queue_set;
+
+// use unnamed namespace to avoid accessing from outside this file
+void StopAndExit(int signal_num) {
+  VLOG(3) << "Catch interrupt signal: " << signal_num << ", program will exit";
+  global_exit_flag = true;
+  for (BlockingQueueSet::iterator iter = global_blocking_queue_set.begin();
+       iter != global_blocking_queue_set.end(); iter++) {
+    iter->get()->Push(
+        std::make_pair(std::string(LISTEN_TERMINATE_MESSAGE), nullptr));
+  }
+  exit(signal_num);
```

Comment:
should exit(0) if success? or else we may think the program have some errors

Reply:
exit(signal_num) will tell the user that their program was exit with the signal sent to them

Reply:
When pserver was killed by kill -9 or kill -TERM or KeyboardInterruption, I guess something in pserver probably went wrong?

Reply:
So I think, the return value should be signal_num rather than 0, to tell the parent process that pserver exits because we stop it manually.

Reply:
But it's OK for me to change it to exit(0), if you take stopping pserver manually as a natural way to stop pserver

Reply:
It's great! thanks very much!